packages/typeahead/src/react/index.tsx

```diff
@@ -135,16 +160,12 @@ const Typeahead: TypeaheadFieldComponent = props => {
             ? { label: searchTerm as string }
             : selectedItem || undefined
         )
-      setInputItems(
-        options.filter(({ label }) =>
-          `${label}`
-            .toLowerCase()
-            // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
-            .startsWith((inputValue as string).toLowerCase())
-        )
-      )
+      setInputItems(filterFunction(options, inputValue))
     }
   })
+  React.useEffect(() => {
+    onChange && onChange(null, activeItem || undefined)
```

Comment:
```suggestion
    onChange && onChange(null, activeItem)
```

`activeItem` can already be undefined and passed as such. Is there another reason to add the `|| undefined`?